test/common/network/connection_impl_test.cc

```diff
@@ -351,9 +357,12 @@ TEST_P(ConnectionImplTest, SetServerTransportSocketTimeout) {
   MockTransportSocket* transport_socket = mocks.transport_socket_.get();
   IoHandlePtr io_handle = std::make_unique<IoSocketHandleImpl>(0);
 
-  auto* mock_timer = new NiceMock<Event::MockTimer>(mocks.dispatcher_.get());
+  auto* mock_timer = new NiceMock<Event::MockTimer>();
```

Comment:
i wonder if this would result into memory leak, wrap in std::make_unique?

Reply:
It gets returned by the EXPECT_CALL on line 363, which wraps it in a unique_ptr.